activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/ResourceModelUtils.java

```diff
@@ -0,0 +1,34 @@
+package com.amazonaws.stepfunctions.cloudformation.activity;
+
+import com.amazonaws.services.stepfunctions.model.DescribeActivityResult;
+import com.amazonaws.services.stepfunctions.model.Tag;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ResourceModelUtils {
+
+    /**
+     * Generates a resource model containing all activity resource properties
+     * @param describeActivityResult The result of calling DescribeActivity for the activity
+     * @param activityTags A list of tags associated with the activity
+     * @return A resource model containing the ARN, Name, and list of tags for the activity
+     */
+    public static ResourceModel getUpdatedResourceModelFromReadResults(DescribeActivityResult describeActivityResult,
```

Comment:
nit: Use `final` keyword for parameters that are not reassigned (which is an anti-pattern to begin with). repeats

Reply:
Updated to have final input parameters. Are you able to expand on the anti-pattern?

Reply:
Not an anti-pattern per se, but a bad practice. It can make code harder to understand as you'd usually expect a function's the inputs to be unmodified throughout unless you're paying close attention.

Here are some explanations:
* https://ncrcoe.gitbooks.io/java-for-small-teams/content/specifics/200_do_not_reassign_parameters.html
* https://eslint.org/docs/rules/no-param-reassign

Reply:
That makes sense, I interpreted the initial comment as final parameters being an anti-pattern. Thanks for the resources! 👍